### DISCUSSION TITLE
Combine datasource information

### DISCUSSION THREAD

**Discussion by astellingwerf** [renovatebot/renovate#24406] [Request Help]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

v36.91.0

### If you're self-hosting Renovate, select which platform you are using.

GitLab self-hosted

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

In the template fields, I see that there's separate for `currentVersion` and `currentDigest`. That makes me think that I can use both to determine what is the next/latest build available.

For Helm charts that I'm developing internally I was thinking of putting them in the version number together: "1.0.2+0badbed0", hoping a `git-refs` datasource would find the latest commit for a branch. But those commits don't have the version number information available in the Helm registry.

Is there a way to specify separate datasource for the version and the digest?

### What you tried so far.

.

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

**Comment by viceice** [comment]
I don't think so

**Reply by astellingwerf** [reply]
Adding a sequential build number to the version helps, because then I can use just the docker datasource to read from GAR, and the regex manager and regex versioning can determine the sequence. But a problem still arises as the artifact is stored via OCI:
```
$ helm push ${HELM_CHART_PACKAGE} ${HELM_CHART_REPO_URL}
Pushed: northamerica-northeast1-docker.pkg.dev/<project>/<repo>/devx:23.4.1_9237.b2ceb3dd
...
northamerica-northeast1-docker.pkg.dev/<project>/<repo>/devx:23.4.1_9237.b2ceb3dd contains an underscore.
OCI artifact references (e.g. tags) do not support the plus sign (+). To support
storing semantic versions, Helm adopts the convention of changing plus (+) to
an underscore (_) in chart version tags when pushing to a registry and back to
a plus (+) when pulling from a registry.
```

The only way that the docker datasource finds the artifact is if the version number it looks for contains an underscore instead of a plus, but the only way the consumer can fetch the artifact is if it uses the plus, not the underscore. How am I supposed to deal with such contradictory conditions?
